modules/library/main/src/main/java/org/geotools/data/store/ArrayDataStore.java

```diff
@@ -25,7 +25,10 @@
 import org.opengis.feature.simple.SimpleFeature;
 import org.opengis.feature.simple.SimpleFeatureType;
 
+@Deprecated
 /**
+ * AbstractDataStore is deprecated. Migrate to ContentDataStore.
```

Comment:
There is now a DataUtilities.store( DataUtilities.source( array ) ) replacement.